kraft-jennifer-ndvi-automation.py

```diff
@@ -0,0 +1,698 @@
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# In[9]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# YOUR CODE HERE
+# Cloud no data vals for Landsat 8 -
+vals = [328, 392, 840, 904, 1350, 352, 368, 416,
+        432, 480, 864, 880, 928, 944, 992, 480, 992]                    #assume these are the same for all
+
+root = os.path.join("ndvi-automation", "sites")
+
+# Open crop boundary
+site_boundary_path = os.path.join(vector_dir,  site_name + "-crop.shp")
+crop_bound = gpd.read_file(site_boundary_path)
+
+dataContainer = []
+for siteName in os.listdir(root):
+    crop_file = os.path.join(root, siteName, 'vector', siteName+'-crop.shp')#build the shp file pathname for this site
+    crop_bound = gpd.read_file(crop_file)                                #build the geopandas representation of the shp
+    
+    baseDir = os.path.join(root, siteName, 'landsat-crop')               #eg) ndvi-automation/sites/HARV/landsat-crop
+    
+    tifBand4 = sorted(glob(os.path.join(baseDir, "LC*","*band*4.tif")))  #fullpath list to band 4 tiffs/sitename
+    tifBand5 = sorted(glob(os.path.join(baseDir, "LC*","*band*5.tif")))  #fullpath list to band 5
+    pixelqa = sorted(glob(os.path.join(baseDir, "LC*","*pixel*.tif")))   #fullpath to qa Raster Mask
+    
+    for b4,b5,qa in zip(tifBand4,tifBand5,pixelqa):                      #zip 2 list together to operate upon
+        date = os.path.split(b4)[1].split('_')[3]                        #4th split object is date in form YYYYmmdd
+        assert (date == os.path.split(b5)[1].split('_')[3])              #learned about assert, making sure times align
```

Comment:
Cool! This is a new one for me - seems like a nice way to take the uncertainty out of the code.